src/main/java/com/bean00/RequestParser.java

```diff
@@ -3,35 +3,41 @@
 import java.io.BufferedReader;
 import java.io.IOException;
 import java.util.ArrayList;
-import java.util.HashMap;
 import java.util.List;
 
 public class RequestParser {
     private BufferedReader in;
-    private HashMap<String, String> requestLineData = new HashMap<>();
 
     public RequestParser(BufferedReader in) {
         this.in = in;
     }
 
     public Request parseRequest() throws IOException {
-        parseRequestLine();
+        String[] requestWords = getRequestWords();
```

Comment:
These helper methods don't seem to provide any value. It seems to just add code without adding any clarity or abstraction. You could inline all three of these methods and loose nothing.

Reply:
Done